benchmarks/benchmarks/signal_filtering.py

```diff
@@ -82,3 +84,54 @@ def setup(self, n_samples, order):
 
     def time_sosfilt_basic(self, n_samples, order):
         sosfilt(self.sos, self.y)
+
+
+class MedFilt2D(Benchmark):
+    def setup_cache(self):
+        np.random.seed(8176)
+        data = np.random.randn(2000, 4000)
+        expected = medfilt2d(data, 5)
+        np.save("medfilt2d_data.npy", data)
+        np.save("medfilt2d_expected.npy", expected)
+
+    def setup(self):
+        self.data = np.load("medfilt2d_data.npy")
+        self.expected = np.load("medfilt2d_expected.npy")
+
+    def teardown(self):
+        assert_allclose(self.output, self.expected)
```

Comment:
Or maybe it would make sense to turn this into a unit test instead. The overlapping chunks do look like they would be a useful test. It should go into `scipy/signal/tests/test_signaltools.py`

Reply:
The size of `data` and `chunks` should again be a lot smaller - if you reduce the size of both axis by a factor then, then it's a nice size for a unit test.